radio/src/maths.cpp

```diff
@@ -172,3 +173,50 @@ void getGpsDistance()
     telemetryData.hub.maxGpsDistance = telemetryData.hub.gpsDistance;
 }
 #endif
+
+#if defined(STM32F4)
+
+inline float degToRad(float x)
+{
+    return x * 0.0174532925f; // Pi / 180
+}
+
+  uint32_t getCoordDistance(int32_t  y1, int32_t x1, int32_t y2, int32_t x2, uint16_t alt) {
+  return(getCoordDistance( (float) y1/1000000, (float) x1/1000000, (float)y2/1000000, (float) x2/1000000, alt));
+}
+
+uint32_t getCoordDistance(float  y1, float x1, float y2, float x2, uint16_t alt){
+  uint32_t dist =  getCoordDistance( y1, x1, y2, x2);
+  return sqrt(dist*dist+alt*alt);
+}
+
+uint32_t getCoordDistance(float  lat1, float long1, float lat2, float long2){
+  //Haversine formula:	a = sin²(Δφ/2) + cos φ1 * cos φ2 * sin²(Δλ/2)
+  //for greek adverse   a = sin²(delta_phi/2) + cos phi1 * cos phi2 * sin²(delta_lambda/2)
+  //                    c = 2 * atan2( √a, √(1−a) )
+  //                    d = R * c
+  // where	φ is latitude, λ is longitude, R is earth’s radius (mean radius = 6,371km); note that angles need to be in radians to pass to trig functions!
+  float R = 6371000.0f; // Earth’s mean, radius in meters
+
+  float phi1 = degToRad(lat1);
+  float phi2 = degToRad(lat2);
+  float delta_phi = degToRad(lat2-lat1);
+  float delta_lambda = degToRad(long2-long1);
```

Comment:
Here you're storing the value of a "cheap" math function (though BTW delta_phi could just be phi2-phi1 to avoid more math converting to radians a 2nd time)...